### PR TITLE
Bump go deps to version 1.8

### PR DESCRIPTION
**Pull request by ReToCode**
## Proposed Changes
- :broom: Bump go deps to version 1.8
- Run `./hack/update-deps.sh --upgrade` and `make generated-files`

/hold

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -22,21 +22,27 @@ require (
...
+	k8s.io/client-go v0.25.4
 	knative.dev/eventing v0.35.2
 	knative.dev/eventing-kafka v0.35.1
 	knative.dev/eventing-kafka-broker v0.35.5
 	knative.dev/hack v0.0.0-20221010154335-3fdc50b9c24a
 	knative.dev/networking v0.0.0-20230224212030-2382b698cb6f
-	knative.dev/operator v0.34.3-0.20230309061722-d32ccab075c6
+	knative.dev/operator v0.35.6
 	knative.dev/pkg v0.0.0-20230224191730-67fb5f6e6735
 	knative.dev/serving v0.34.1
```
Comment (matzew) [inline go.mod:35]: this still points to 0.34?

File: hack/update-deps.sh [hunk 2]
```diff
@@ -13,8 +13,8 @@ set -o pipefail
 cd "${ROOT_DIR}"
 
 # This controls the knative release version we track.
-KN_VERSION="release-1.7"
-EVENTING_VERSION="release-v1.7"
+KN_VERSION="release-1.8"
+EVENTING_VERSION="release-v1.8"
```
Comment (matzew) [inline hack/update-deps.sh:17]: I wonder, now, should that have been part of my eventing bump?

File: hack/update-deps.sh [hunk 3]
```diff
@@ -13,8 +13,8 @@ set -o pipefail
 cd "${ROOT_DIR}"
 
 # This controls the knative release version we track.
-KN_VERSION="release-1.7"
-EVENTING_VERSION="release-v1.7"
+KN_VERSION="release-1.8"
```
Comment (matzew) [inline hack/update-deps.sh:16]: Do we really want to biump `kn` version too, at this point? /cc @dsimansk

File: go.mod [hunk 4]
```diff
@@ -22,21 +22,27 @@ require (
...
+	k8s.io/api v0.25.4
+	k8s.io/apimachinery v0.25.4
+	k8s.io/client-go v0.25.4
 	knative.dev/eventing v0.35.2
 	knative.dev/eventing-kafka v0.35.1
 	knative.dev/eventing-kafka-broker v0.35.5
 	knative.dev/hack v0.0.0-20221010154335-3fdc50b9c24a
 	knative.dev/networking v0.0.0-20230224212030-2382b698cb6f
-	knative.dev/operator v0.34.3-0.20230309061722-d32ccab075c6
+	knative.dev/operator v0.35.6
```
[the remark this answers is not in the document]
Comment (ReToCode) [reply go.mod:33]: I think its generated out of the branch we specify in `project.yaml`, but this one is actually pretty use-less, as it is overwritten by the replace below `knative.dev/operator => knative.dev/operator v0.32.2`.

[the remark this answers is not in the document]
Comment (ReToCode) [reply go.mod:33]: I'm usually just looking up the tags and check for the same commit-id: https://github.com/knative/operator/tags.

File: go.mod (hunk 1 above)
Comment (ReToCode) [reply go.mod:35]: Stavros is working on a PR to bump operator.

File: hack/update-deps.sh (hunk 3 above)
Comment (ReToCode) [reply hack/update-deps.sh:16]: KN here, amusingly, does not mean the client, it is used in various places below like 

```
  "knative.dev/pkg=knative.dev/pkg@${KN_VERSION}"
  "knative.dev/hack=knative.dev/hack@${KN_VERSION}"
  "knative.dev/networking=knative.dev/networking@${KN_VERSION}"
```

Comment (matzew) [reply hack/update-deps.sh:16]: Ah, great. LOL

File: go.mod (hunk 1 above)
Comment (ReToCode) [reply go.mod:35]: That's wrong and will be fixed.

**Comment by nak3** [comment]
/lgtm

`ci/prow/4.10-upstream-e2e-mesh-aws-ocp-410` is failing but it will be fixed by: https://github.com/openshift-knative/net-istio/pull/26  https://github.com/openshift-knative/net-istio/pull/27

**Comment by ReToCode** [comment]
Thank you @nak3 . I'd like to have one build with all tests (including not required) passing. So I'll re-generate the manifests here as soon as those are merged.

**Pull request merged**